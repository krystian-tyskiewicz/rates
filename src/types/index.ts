export type Rate = {
  currency: string
  code: string
  mid: number
}
