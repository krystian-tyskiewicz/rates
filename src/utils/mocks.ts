export const rates = [
  {
    currency: 'ariary (Madagaskar)',
    code: 'MGA',
    mid: 0.000885,
  },
  {
    currency: 'balboa (Panama)',
    code: 'PAB',
    mid: 4.0347,
  },
]
