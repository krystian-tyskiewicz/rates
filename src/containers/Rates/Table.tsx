import Table from '@mui/material/Table'
import TableBody from '@mui/material/TableBody'
import TableCell from '@mui/material/TableCell'
import TableHead from '@mui/material/TableHead'
import TableRow from '@mui/material/TableRow'
import Paper from '@mui/material/Paper'

import { Rate } from 'types'

type Props = {
  rates: Rate[]
}

const RatesTable = ({ rates }: Props) => {
  return (
    <Paper sx={{ overflowX: 'auto' }}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Currency</TableCell>
            <TableCell>Code</TableCell>
            <TableCell>Mid</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rates.map((rate) => (
            <TableRow key={rate.code}>
              <TableCell>{rate.currency}</TableCell>
              <TableCell>{rate.code}</TableCell>
              <TableCell>{rate.mid}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  )
}

export default RatesTable
