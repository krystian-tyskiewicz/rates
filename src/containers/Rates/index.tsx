import { styled } from '@mui/material'
import CircularProgress from '@mui/material/CircularProgress'

import { Rate } from 'types'
import {
  useGetRatesQuery,
} from 'redux/services/rates'
import RatesTable from './Table'

const Progress = styled(CircularProgress)`
  align-self: center;
`

const RatesListPage = () => {
  const { data: rates, isLoading } = useGetRatesQuery()

  if (isLoading) {
    return <Progress />
  }

  return (
    <RatesTable
      rates={rates as Rate[]}
    />
  )
}

export default RatesListPage
