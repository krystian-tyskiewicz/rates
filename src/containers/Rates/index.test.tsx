import { render } from '@testing-library/react'

import Rates from '.'
import { rates as mockRates } from 'utils/mocks'
import { Rate } from 'types'

jest.mock('redux/services/rates', () => ({
  useGetRatesQuery: jest.fn().mockImplementation(() => ({
    data: mockRates,
  })),
}))

jest.mock('./Table', () => {
  const Table = (rates: Rate[]) => (
    <div className="table">{JSON.stringify(rates)}</div>
  )
  return Table
})

it('renders correctly', () => {
  const { container } = render(<Rates />)
  expect(container).toMatchSnapshot()
})
