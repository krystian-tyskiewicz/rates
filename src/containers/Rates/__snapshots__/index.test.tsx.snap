// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="table"
  >
    {"rates":[{"currency":"ariary (Madagaskar)","code":"MGA","mid":0.000885},{"currency":"balboa (Panama)","code":"PAB","mid":4.0347}]}
  </div>
</div>
`;
